custom_components/variable/helpers.py

```
@@ -120,162 +132,115 @@ def to_num(s):
+) -> str | float | datetime.date | datetime.datetime:
+    """Convert a date to the requested variable value type."""
+    if dest_type is None or dest_type == "string":
+        return value.isoformat()
+    if dest_type == "date":
+        return value
+    combined = datetime.datetime.combine(value, datetime.time.min)
+    if dest_type == "datetime":
+        return combined
+    if dest_type == "number":
+        return combined.timestamp()
+    raise ValueError(f"Invalid dest_type: {dest_type}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)